refactor: move uptime tracking from system_status_server(HTTP) to DRT level

keivenchang opened the pull request:
#### Overview:

Refactors system metrics and uptime tracking architecture to improve separation of concerns (e.g. the uptime should NOT be dependent on enabling the system_status_server/HTTP).

#### Details:

- Move uptime tracking from SystemStatusState to DRT's SystemHealth
- Improve test isolation and reduce component coupling
- Consolidate metrics initialization in DistributedRuntime
- Simplify SystemStatusState by removing uptime responsibilities

#### Where should the reviewer start?

- `lib/runtime/src/lib.rs` - New SystemHealth uptime implementation
- `lib/runtime/src/system_status_server.rs` - Simplified SystemStatusState
- `lib/runtime/src/distributed.rs` - Metrics initialization

#### Related Issues:

Relates to DIS-482

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- New Features
  - Added automatic uptime tracking with an exported uptime metric initialized at startup.
  - Uptime is continuously updated and reported in seconds.

- Bug Fixes
  - Uptime metrics now work even when system status endpoints are disabled.
  - Duplicate metric registration is handled gracefully without failures.

- Refactor
  - Centralized uptime logic in the system health component for consistent reporting across health and metrics endpoints.
  - Simplified status server to rely on centralized health data.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

keivenchang committed:
refactor: consolidate system metrics and uptime tracking architecture

rmccorm4 commented on lib/runtime/src/distributed.rs:
```diff
@@ -153,17 +160,7 @@ impl DistributedRuntime {
-                tracing::warn!("Failed to initialize system status start time: {}", e);
-            }
-
+            // System server HTTP is disabled, but uptime metrics are still being tracked via SystemHealth
```
Does it make sense to add a small test case that basically does this?
```bash
DYN_SYSTEM_ENABLED=false
assert <request to system status server fails as expected because it's not running>
```

This will be a signal to other developers that something is wrong if they add the server back in this `else` block in future, and catch subtle things like if we change the mechanism or name of the environment variable for toggling support or something.

rmccorm4 reviewed (approved):
one comment, LGTM otherwise - please fix the merge conflict too

keivenchang committed:
refactor: consolidate uptime tracking in SystemHealth and improve thread safety

- Move uptime tracking from SystemStatusState to SystemHealth
- Replace Option<prometheus::Gauge> with OnceLock<prometheus::Gauge> for better thread safety
- Add tests for uptime functionality with system enabled/disabled
- Fix clippy warning by removing unnecessary ref in pattern matching
